src/test/java/alerts/FundsAlertTest.java

```diff
@@ -0,0 +1,130 @@
+        
+        String output = outputStream.toString();
+        assertTrue(output.contains("WARNING") && output.contains("below warning threshold"),
+                "Alert should be triggered when funds are below threshold");
+    }
+    
+    @Test
+    public void update_fundsAboveThreshold_expectNoAlert() {
+        outputStream.reset();
+        
+        fundsAlert.update(10.0, 100.0, 90.0, 0.0);
+        
+        String output = outputStream.toString();
+        assertFalse(output.contains("WARNING") && output.contains("below warning threshold"),
+                "No alert should be triggered when funds are above threshold");
+    }
+    
+    @Test
+    public void displayInitialNotification_defaultSettings_expectCorrectMessage() {
+        outputStream.reset();
+        
+        fundsAlert.displayInitialNotification();
+        
+        String output = outputStream.toString();
+        assertTrue(output.contains("Funds Alert feature is active"),
+                "Initial notification should indicate the feature is active");
+        assertTrue(output.contains("below $5.00"),
+                "Initial notification should show the default threshold");
+        assertTrue(output.contains("alert set"),
+                "Initial notification should mention how to change the threshold");
+    }
+    
+    @Test
+    public void update_thresholdChangedAndFundsCrossThreshold_correctAlertBehavior() throws BudgetTrackerException {
+        // First set a higher threshold
+        fundsAlert.setWarningThreshold(20.0);
+        
+        // Test with funds above new threshold - should not trigger alert
+        outputStream.reset();
+        fundsAlert.update(25.0, 100.0, 75.0, 0.0);
+        String output1 = outputStream.toString();
+        assertFalse(output1.contains("WARNING"), 
+                "No alert should be triggered when funds are above the new threshold");
+        
+        // Test with funds below new threshold but above original threshold - should trigger alert
+        outputStream.reset();
+        fundsAlert.update(15.0, 100.0, 85.0, 0.0);
+        String output2 = outputStream.toString();
+        assertTrue(output2.contains("WARNING") && output2.contains("below warning threshold"),
+                "Alert should be triggered when funds are below the new threshold");
+        assertTrue(output2.contains("$15.00") && output2.contains("$20.00"),
+                "Alert should show the correct available funds and threshold values");
+        
+        // Reset threshold to lower value
+        fundsAlert.setWarningThreshold(10.0);
+        
+        // Same funds should now be above threshold - should not trigger alert
+        outputStream.reset();
+        fundsAlert.update(15.0, 100.0, 85.0, 0.0);
+        String output3 = outputStream.toString();
+        assertFalse(output3.contains("WARNING"),
+                "No alert should be triggered when threshold is lowered below the available funds");
+    }
```

Comment:
you can move some of these into @before and @after methods if you wish to keep the tests clean and remove somewhat repetitive code